src/config/enterprise.rs

```diff
@@ -103,6 +103,47 @@ impl Default for Options {
     }
 }
 
+impl Options {
+    pub fn merge(&mut self, other: Self) {
+        self.enabled = other.enabled;
+        self.enable_logs_reporting = other.enable_logs_reporting;
+        if other.site.is_some() {
+            self.site = other.site;
+        }
+        if other.region.is_some() {
+            self.region = other.region;
+        }
+        if other.endpoint.is_some() {
+            self.endpoint = other.endpoint;
+        }
+        if other.api_key.is_some() {
+            self.api_key = other.api_key;
```

Comment:
Similarly to above, it feels like we could at least log here for transparency if the values differ.